hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/internal/DataSourceInternalWriterHelper.java

```diff
@@ -68,6 +68,7 @@ public DataSourceInternalWriterHelper(String instantTime, HoodieWriteConfig writ
     this.metaClient = HoodieTableMetaClient.builder().setConf(configuration).setBasePath(writeConfig.getBasePath()).build();
     this.metaClient.validateTableProperties(writeConfig.getProps());
     this.hoodieTable = HoodieSparkTable.create(writeConfig, new HoodieSparkEngineContext(new JavaSparkContext(sparkSession.sparkContext())), metaClient);
+    writeClient.preWrite(instantTime, WriteOperationType.BULK_INSERT, metaClient);
```

Comment:
The fix makes sense but the not a good idea to put a write logic in the constructor, WDYT ?

Reply:
yeah. makes sense. Only other option I see is to add it to createInflightCommit(). Do you suggest to do that or should we introduce a new method and explicitly call before calling createInflightCommit?

Reply:
it is not ideal but sort of existing pattern. `writeClient.startCommitWithTime` is already writing files there. I'd say it's ok to go ahead with this and file a refactoring task

Reply:
https://issues.apache.org/jira/browse/HUDI-4444 filed